project-base/app/src/DataFixtures/Demo/DemoDataFactory/ProductDemoDataFactory.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\DataFixtures\Demo\DemoDataFactory;
+
+use App\DataFixtures\Demo\UnitDataFixture;
+use App\DataFixtures\Demo\VatDataFixture;
+use App\Model\Product\ProductData;
+use App\Model\Product\ProductDataFactory;
+use App\Model\Product\Unit\Unit;
+use DateTime;
+use Shopsys\FrameworkBundle\Component\DataFixture\PersistentReferenceFacade;
+use Shopsys\FrameworkBundle\Component\Domain\Domain;
+use Shopsys\FrameworkBundle\Model\Pricing\Vat\Vat;
+
+class ProductDemoDataFactory
+{
+    /**
+     * @param \Shopsys\FrameworkBundle\Component\Domain\Domain $domain
+     * @param \App\Model\Product\ProductDataFactory $productDataFactory
+     * @param \Shopsys\FrameworkBundle\Component\DataFixture\PersistentReferenceFacade $persistentReferenceFacade
+     */
+    public function __construct(
+        private readonly Domain $domain,
+        private readonly ProductDataFactory $productDataFactory,
+        private readonly PersistentReferenceFacade $persistentReferenceFacade,
+    ) {
+    }
+
+    /**
+     * @return \App\Model\Product\ProductData
+     */
+    public function createDefaultData(): ProductData
+    {
+        $productData = $this->productDataFactory->create();
+
+        $this->setVat($productData, VatDataFixture::VAT_HIGH);
+        $this->setUnit($productData, UnitDataFixture::UNIT_PIECES);
+        $this->setSellingTo($productData, null);
+
+        $productData->sellingDenied = false;
+
+        return $productData;
+    }
+
+    /**
+     * @param \App\Model\Product\ProductData $productData
+     * @param string|null $vatReference
+     */
+    private function setVat(ProductData $productData, ?string $vatReference): void
+    {
+        $productVatsIndexedByDomainId = [];
+
+        foreach ($this->domain->getAllIds() as $domainId) {
+            if ($vatReference !== null) {
+                $productVatsIndexedByDomainId[$domainId] = $this->persistentReferenceFacade->getReferenceForDomain($vatReference, Domain::FIRST_DOMAIN_ID, Vat::class);
```

Comment:
After the refactoring is done, I would like to fix this line - there is a mistake - we assign the first domain Vat for all the domains. I am not sure whether this will break some tests or not, we will see :smile: :see_no_evil:

Reply:
fixed, lets see what will be reported from tests 😅

Reply:
you will need to rebase the branch first :slightly_smiling_face: